docs/metrics.md

```diff
@@ -84,6 +84,12 @@ The total number of VMs created by namespace and virt-api pod, since install. Ty
 ### kubevirt_vm_created_total
 The total number of VMs created by namespace, since install. Type: Counter.
 
+### kubevirt_vm_disk_allocated_size_bytes
+Disks allocated size in bytes. Type: Gauge.
+
+### kubevirt_vm_disk_used_size_bytes
```

Comment:
How do you plan to get this data?  Do you need guest agent integration?

Reply:
This query returns the used size by volume name, namespace, but its missing the connection to the vm name.

sum by (name, namespace)(kubelet_volume_stats_used_bytes + on (persistentvolumeclaim,namespace) group_left(name)(label_replace(0*kube_persistentvolumeclaim_info , "persistentvolume",  "$1",  "volumename",  "(.*)") +on (persistentvolume) group_left(name)(0*kube_persistentvolume_claim_ref)))


so to be honest I'm not sure yet if I should use this metric and somehow add the vm name or get the used size directly. any ideas?

Reply:
I can't imagine that query works for block PVCs.

I don't think there is any standard way you can get used bytes for a block device from k8s. You could get free space from guest perspective but if user "preallocated" the DataVolume than actual used will likely be greater

For filesystem PVCs, you can stat the disk.img file to get accurate used/free byte stats that will reflect preallocation

Reply:
ohhh so that's a problem.. thanks for the info.

@sradco WDYT?

Reply:
something like https://libvirt.org/manpages/virsh.html#domfsinfo would work?

Reply:
@mhenriks WDYT about joao comment?